.ci/check_python_dists.sh

```diff
@@ -17,4 +17,35 @@ if { test "${TASK}" = "bdist" || test "${METHOD}" = "wheel"; }; then
     check-wheel-contents ${DIST_DIR}/*.whl || exit -1
 fi
 
+PY_MINOR_VER=$(python -c "import sys; print(sys.version_info.minor)")
+if [ $PY_MINOR_VER -gt 7 ]; then
+    echo "pydistcheck..."
+    pip install pydistcheck
+    if [ $TASK == "CUDA" ] && [ $METHOD == "wheel" ]; then
+        pydistcheck \
+            --inspect \
+            --ignore 'compiled-objects-have-debug-symbols,max-allowed-size-compressed' \
+            --max-allowed-size-uncompressed '60M' \
+            --max-allowed-files 800 \
+            ${DIST_DIR}/* || exit -1
```

Comment:
This `if` branch accounts for the fact that the CUDA 11.x wheel currently has the following warnings from `pydistcheck`:

```text
1. [compiled-objects-have-debug-symbols] Found compiled object containing debug symbols. For details, extract the distribution contents and run 'nm --debug-syms "lightgbm/lib_lightgbm.so"'.
2. [distro-too-large-compressed] Compressed size 35.1M is larger than the allowed size (5.0M).
3. [distro-too-large-uncompressed] Uncompressed size 54.8M is larger than the allowed size (15.0M).
errors found while checking: 3
```

([build link](https://github.com/microsoft/LightGBM/actions/runs/4739461720/jobs/8414327804?pr=5838))

I'm not sure why the CUDA build is so much larger than the non-CUDA one or where those debug symbols are coming from...maybe we are statically linking to CUDA libraries? I'm not sure.

Either way, I'm proposing not spending time investigating that right now, and instead just at least adding this check to detect if changes make the wheel significantly larger (which could be a problem in storage-sensitive environments like AWS Lambda).